### PR TITLE
Fix untested doctest phen_micro

### PR DESCRIPTION
**Pull request by dizak**
This pull request addresses untested doctests of ```Bio.phenotype.phen_micro```

- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

The tests were run on TravisCi activated for my fork of this repository. For this purpose, the ```Tests.run_tests.py``` and ```.travis-tox.ini``` were modified. Otherwise, the tests would be skipped since this module requires ```numpy```. I suppose the modifications of ```.travis-tox.ini``` should be reverted once the rest of the changes in this pull request are accepted.

The tests failed for Python 3.8 due to unsuccessful installation of ```numpy```. It seems it is independent of the doctests themselves.

### CODE REVIEW

**Comment by MarkusPiotrowski** [comment]
@dizak The files which are listed in the 'numpy-requiring' section of `run_tests.py` don't need to be addressed. They need to stay there to allow running the test suite locally even if `numpy` is not installed. 
Actually, the doctests of `phen_micro.py` seemed to be fine before:

```bash
> python run_tests.py docstest
...
Bio.phenotype docstring test ... ok
Bio.phenotype.phen_micro docstring test ... ok
Bio.phenotype.pm_fitting docstring test ... skipped, missing Python dependency
...
```
If `numpy` is installed, the doctests of `phen_micro.py` are started and don't make any issues. Thus it seems that your changes were not necessary. The three dots `...` in the doctests output are called ELLIPSIS and they match any output, so it is not necessary to reproduce the whole output, e.g. a larger list etc.

File: Bio/phenotype/phen_micro.py [hunk 1]
```diff
@@ -112,18 +109,29 @@ class PlateRecord(object):
...
+    H10
+    H11
+    H12
 
     All the wells belonging to a "column" (identified by the number of the well)
     in the plate can be obtained:
 
     >>> for well in plate.get_column(12):
     ...     print(well.id)
-    ...
```
Comment (peterjc) [inline Bio/phenotype/phen_micro.py]: Can you put this ``...`` line back, you would see it at the interactive python prompt.

File: Bio/phenotype/phen_micro.py [hunk 2]
```diff
@@ -112,18 +109,29 @@ class PlateRecord(object):
     H01
     H02
     H03
-    ...
```
Comment (peterjc) [inline Bio/phenotype/phen_micro.py]: Why fill out the full list? You could list 1, 2, 3, ..., 12 if you wanted to be a little more explicit.

File: Tests/run_tests.py [hunk 3]
```diff
@@ -128,7 +128,6 @@ def is_pypy():
         "Bio.PDB.Vector",
         "Bio.phenotype",
         "Bio.phenotype.parse",
-        "Bio.phenotype.phen_micro",
```
Comment (peterjc) [inline Tests/run_tests.py]: Why are you removing this check? Don't the ``Bio.phenotype.phen_micro`` tests require NumPy?

**Comment by peterjc** [comment]
Markus is right, e.g. a very recent master branch test, looking at Python 3.7 where numpy is installed:

https://travis-ci.org/biopython/biopython/jobs/578252955

```
...
test_pairwise_aligner ... ok
test_phenotype ... ok
test_phenotype_fit ... ok
test_phyml_tool ... ok
test_prodoc ... ok
...
Bio.pairwise2 docstring test ... ok
Bio.phenotype docstring test ... ok
Bio.phenotype.phen_micro docstring test ... ok
Bio.phenotype.pm_fitting docstring test ... ok
Bio.triefind docstring test ... ok
...
```

I don't see anything which needs fixing here?

**Comment by dizak** [comment]
All your remarks are correct. I am sorry I wasted your time.

The first changes were made after running the doctests manually, without the ```ELLIPSIS``` flag. Running the tests with ```Tests/run_tests.py``` goes as expected, since:

https://github.com/biopython/biopython/blob/fd4d1866facc2185ebca3b2cd05a47a47f4ecacd/Tests/run_tests.py#L364-L365

I should see this before. Apologies once more, this can be closed.
Please mark the ```Bio.phenotype.phen_micro``` as done #1902

**Comment by peterjc** [comment]
Ticked on #1902, thanks for spotting that oversight.

When you said you ran the doctests manually, what did you mean?

You can also run an individual file's doctests via this snippet at the end of the file:

```python
if __name__ == "__main__":
    from Bio._utils import run_doctest
    run_doctest(verbose=0)
```

(However we've found for some modules this does not work on Python 3 due to local imports)

**Comment by dizak** [comment]
In my stupidity, I have just run ```python -m doctest phen_micro.py -v```. If run like this, the ```doctest``` treats ```ELLIPSE``` literally. I did my reading later (too late).

**Comment by peterjc** [comment]
Ah, in that case ``python -m doctest -o ELLIPSIS phen_micro.py -v`` ought to work? Thanks for clarifying, and do let us know if our testing documentation could be clarified (especially the chapter in the tutorial).

**Comment by dizak** [comment]
I believe the documentation is fine, it was just the #1902 TODO that made me think there **must** be something wrong with it since it's on the list :)

**Pull request closed without merging**